components/core/header/nav-bar/ScrollableNavBar.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+    <div class="core-scrollableNavBar">
+        <button
+            class="core-scrollableNavBar__left-arrow"
+            @click="onLeftArrowClick"
+        >
+            <span
+                :class="{
+                    'core-scrollableNavBar__arrow-clickable': leftClickable,
+                }"
+            >
+                <fa :icon="['fa', 'chevron-left']" aria-hidden="true"></fa>
+            </span>
+        </button>
+        <button
+            class="core-scrollableNavBar__right-arrow"
+            @click="onRightArrowClick"
+        >
+            <span
+                :class="{
+                    'core-scrollableNavBar__arrow-clickable': rightClickable,
+                }"
+            >
+                <fa :icon="['fa', 'chevron-right']" aria-hidden="true"></fa>
+            </span>
+        </button>
+        <div
+            ref="scrollableNavBarScroll"
+            class="
+                core-scrollableNavBar__scroll core-scrollableNavBar__overflow
+            "
+        >
+            <div class="core-scrollableNavBar__slot">
+                <slot></slot>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+export default {
+    name: 'CoreScrollableNavBar',
+    props: {
+        scrollDelta: {
+            default: 1000,
+            type: Number,
+        },
+    },
+    data() {
+        return {
+            leftClickable: false,
+            rightClickable: true,
+        }
+    },
+    mounted() {
+        const el = this.$refs.scrollableNavBarScroll
+        this.leftClickable = !this.checkHorizontalScrollStart(el)
+        this.rightClickable = !this.checkHorizontalScrollEnd(el)
+    },
+    methods: {
+        checkHorizontalScrollStart(el, scrollLeft) {
+            if (scrollLeft === undefined) {
+                scrollLeft = el.scrollLeft
+            }
+            return scrollLeft <= 0
+        },
+        checkHorizontalScrollEnd(el, scrollLeft) {
+            if (scrollLeft === undefined) {
+                scrollLeft = el.scrollLeft
+            }
+            return el.scrollWidth - el.clientWidth - scrollLeft < 1
+        },
+        horizontalScroll(delta) {
+            const el = this.$refs.scrollableNavBarScroll
+            const newScrollleft = el?.scrollLeft + delta
```

Comment:
1. Use proper camelCase to avoid variable name typos - `newScrollLeft`
2. If `el?.scrollLeft` evaluates as `undefined`, then is the value of `newScrollLeft` `undefined` or `NaN`?

Reply:
88493a9